src/ruby/ArrayConcepts.java

```diff
@@ -0,0 +1,78 @@
+/*write a single program having different methods.
+ 
+a) To find Min number from given array
+b) To find Max number from given array
+c) To find Avg of all numbers from give array
+d) To display all numbers [0 to length]
+e) To display all numbers [length to 0]
+g) Method to find average of min and max number from array.*/
+
+package ruby;
+
+public class ArrayConcepts {
+	
+	public static void main(String[] args) {
+		int[] num = {10,60,30,40,50};
+		ArrayConcepts arr = new ArrayConcepts();
+		
+		int max = arr.findMax(num);
+		int min = arr.findMin(num);
+		
+		int average = arr.average(num);
+		System.out.println("Average of Array is "+average);
+		
+		arr.averageOfMinMax(max, min);
+		
+		System.out.println("Display Array List from 0 to "+num.length);
+		arr.displayArraylist(num);
+		
+		System.out.println("Display Array List in Reverse order from "+num.length+ "to 0");
+		arr.displayReverseArraylist(num);
+	}
+	
+	void averageOfMinMax(int num1, int num2){
```

Comment:
I think we have requirement where you need to find max & min number then you need to find avg of both, so this expecting two variable in method looks wrong as per the implementation perspective, user will give us array, not min & max value. I hope you got my point.


You can write code like this, I remove variable and directly doing sum of min & max, no problem, if you are not comfortable to write code like below, just optimize.

int avgOfMinMax = (findMax(input)+findMin(input))/2